File: experiments/llm-reasoning-cost/generated_code/go/02_fibonacci_codex.go
package main

import "fmt"

func fib(n int32) int32 {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	fmt.Println(fib(0))
	fmt.Println(fib(1))
	fmt.Println(fib(5))
	fmt.Println(fib(10))
}

File: experiments/llm-reasoning-cost/generated_code/rust/02_fibonacci_codex.rs
use std::io::{self, Read};

fn fib(n: i32) -> i32 {
    if n < 2 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let n: i32 = input.trim().parse().unwrap_or(0);
    let result = fib(n);
    println!("{}", result);
}